report_compassion/models/contract_group.py

```diff
@@ -172,6 +172,10 @@ def get_communication(self, start, stop, sponsorships):
             elif number_sponsorship and valid.child_id:
                 vals['subject'] = valid.child_id.firstname + " ({})".format(
                     valid.child_id.local_id)
+            elif number_sponsorship and not valid.child_id and valid.display_name:
+                product_name = self.env['product.product'].search([('id', '=', valid.contract_line_ids.product_id.id)])
+                vals['subject'] = product_name.name
```

Comment:
Use `", ".join(product_name.mapped('thanks_name'))`
As there can be several products and thanks_name is a more user friendly name of the product